packages/bitcore-node/test/integration/models/block.integration.ts

```diff
@@ -0,0 +1,611 @@
+          time: 1526326785,
+          version: '536870912',
+          merkleRoot: '8c29860888b915715878b21ce14707a17b43f6c51dfb62a1e736e35bc5d8093f',
+          bits: parseInt('207fffff', 16).toString(),
+          nonce: '3'
+        },
+        chain: 'BTC',
+        network: 'regtest'
+      });
+
+      const result = await BlockModel.find({ chain: 'BTC', network: 'regtest'}).exec();
+      expect(result.length).to.equal(3);
+
+    });
+    it('should return if localTip height is zero', async () => {
+      await BlockModel.create({
+        chain: 'BTC',
+        network: 'regtest',
+        height: 1335,
+        hash: '528f01c17829622ed6a4af51b3b3f6c062f304fa60e66499c9cbb8622c8407f7',
+        version: '536870912',
+        merkleRoot: 'a2262b524615b6d2f409784ceff898fd46bdde6a584269788c41f26ac4b4919e',
+        time: 1526326784,
+        nonce: '3',
+        previousBlockHash: '64bfb3eda276ae4ae5b64d9e36c9c0b629bc767fb7ae66f9d55d2c5c8103a929',
+        size: 264,
+        bits: parseInt('207fffff', 16).toString(),
+        processed: true
+      });
+      await BlockModel.create({
+        chain: 'BTC',
+        network: 'regtest',
+        height: 1336,
+        hash: '2a883ff89c7d6e9302bb4a4634cd580319a4fd59d69e979b344972b0ba042b86',
+        version: '536870912',
+        merkleRoot: '8a351fa9fc3fcd38066b4bf61a8b5f71f08aa224d7a86165557e6da7ee13a826',
+        time: 1526326785,
+        nonce: '0',
+        previousBlockHash: '528f01c17829622ed6a4af51b3b3f6c062f304fa60e66499c9cbb8622c8407f7',
+        size: 264,
+        bits: parseInt('207fffff', 16).toString(),
+        processed: true
+      });
+      await BlockModel.create({
+        chain: 'BTC',
+        network: 'regtest',
+        height: 1337,
+        hash: '3279069d22ce5af68ef38332d5b40e79e1964b154d466e7fa233015a34c27312',
+        version: '536870912',
+        merkleRoot: '8a351fa9fc3fcd38066b4bf61a8b5f71f08aa224d7a86165557e6da7ee13a826',
+        time: 1526326785,
+        nonce: '3',
+        previousBlockHash: '2a883ff89c7d6e9302bb4a4634cd580319a4fd59d69e979b344972b0ba042b86',
+        size: 264,
+        bits: parseInt('207fffff', 16).toString(),
+        processed: true,
+      });
+
+      await BlockModel.handleReorg({
+        header: {
+          prevHash: '12c719927ce18f9a61d7c5a7af08d3110cacfa43671aa700956c3c05ed38bdaa',
+          hash: '4c6872bf45ecab2fb8b38c8b8f50fc4a8309c6171d28d479b8226afcb1a99920',
+          time: 1526326785,
+          version: '536870912',
+          merkleRoot: '8c29860888b915715878b21ce14707a17b43f6c51dfb62a1e736e35bc5d8093f',
+          bits: parseInt('207fffff', 16).toString(),
+          nonce: '3'
+        },
+        chain: 'ETH',
+        network: 'regtest'
+      });
+
+      const result = await BlockModel.find({ chain: 'BTC', network: 'regtest'}).exec();
+      expect(result.length).to.equal(3);
+
+    });
+    it('should successfully handle reorg', async () => {
```

Comment:
we should add comments to this test to explain what we are doing and checking in the various parts

Reply:
updated

Reply:
confirmed